uni/lib/model/entities/exam.dart

```diff
@@ -1,31 +1,57 @@
 import 'package:intl/intl.dart';
+import 'package:json_annotation/json_annotation.dart';
+import 'package:logger/logger.dart';
 import 'package:uni/model/entities/app_locale.dart';
 
+part '../../generated/model/entities/exam.g.dart';
+
+class DateTimeConverter extends JsonConverter<DateTime, String> {
+  const DateTimeConverter();
+
+  @override
+  DateTime fromJson(String json) {
+    final format = DateFormat('yyyy-M-dd');
+    return format.parse(json);
+  }
+
+  @override
+  String toJson(DateTime object) {
+    final format = DateFormat('yyyy-MM-dd');
+
+    return format.format(object);
+  }
+}
```

Comment:
Same as the other converter, this should be a file itself. 

Also, we only store the date and not the time, which I think will lead to problems as the exam time is also stored inside a DateTime, and the format between `fromJson` and `toJSON` is different. You should use the [toIso8601String](https://api.flutter.dev/flutter/dart-core/DateTime/toIso8601String.html) function, and parse it with the `DateTime.parse` static method. This should make the converter universal in all use cases.